### PR TITLE
feat: EIP-1271 accordion w/ summary

### LINKED ISSUE #1156: [EIP-1271] Accordion summary
## What is the feature about

Implementation of EIP-1271 "Messages" accordion.

## The list of requirements

- [x] Create stubbed `useMessages` hook that returns mocked data
- [x] Create accordion (reusing the CSS of the transaction list) for messages
- [x] Create `MsgSummary`, similar to `TxSummary`

Note: the "details" and "stepper" will be tackled in separate PRs, found in the epic [here](https://github.com/safe-global/web-core/issues/1088).

## Designs/sketches

image

## Links

- [Problem statement/data structure](https://www.notion.so/gnosis-safe/EIP-1271-signatures-can-only-be-generated-on-chain-2800b5e516124d959c38ba8f2042f9f7?d=e8ca1d332df54ae587e66240fd40e230#a8489e3724234f75a9c3d389e323a725)
- [Figma design](https://www.figma.com/file/x1uuIJsOZkyv5cwNlDdVC1/EIP-1271?node-id=1%3A2&t=F7CHK05WWFuPl1sq-0)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #1156

## How this PR fixes it

An expandable `Accordion` that uses the same styles as that from the transaction list has been created, with a "summary" that is of the same manner as that of transactions.

## How to test it

Open "Messages" in the transaction list and obeserve the new accordions. Opening `/transactions/msg?id={{messageHash}}&safe={{shortName}}:{{address}}` should expand the accordion by default.

## Screenshots

image

image

### CODE REVIEW

**Comment by iamacook** [comment]
As we have yet to plan the individual tasks, I will leave this as a draft. We will need to refactor some of the components used here as they are almost identical to those from the transaction list. I suggest we tackle the rest of this tomorrow. cc @usame-algan

File: src/components/messages/SignMsgButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import Track from '@/components/common/Track'
+import { MESSAGE_EVENTS } from '@/services/analytics/events/txList'
+import useIsMessageSignableBy from '@/hooks/useIsMsgSignableBy'
+import useIsMsgPending from '@/hooks/useIsMsgPending'
+import type { Message } from '@/hooks/useMessages'
+
+const SignMsgButton = ({ msg, compact = false }: { msg: Message; compact?: boolean }): ReactElement => {
+  const wallet = useWallet()
+  const isSignable = useIsMessageSignableBy(msg, wallet?.address || '')
+  const isSafeOwner = useIsSafeOwner()
```
Comment (schmanu) [inline src/components/messages/SignMsgButton/index.tsx:17]: `useIsMessageSignableBy` already checks for `useIsSafeOwner`

File: src/hooks/useIsMsgSignableBy.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import useIsSafeOwner from '@/hooks/useIsSafeOwner'
+import type { Message } from '@/hooks/useMessages'
+
+const useIsMessageSignableBy = (message: Message, walletAddress: string): boolean => {
+  const isSafeOwner = useIsSafeOwner()
+  return isSafeOwner ? message.confirmations.every(({ owner }) => owner.value !== walletAddress) : false
```
Comment (schmanu) [inline src/hooks/useIsMsgSignableBy.ts:6]:
```suggestion
  return isSafeOwner && message.confirmations.every(({ owner }) => owner.value !== walletAddress)
```

File: src/hooks/useIsMsgSignableBy.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import useIsSafeOwner from '@/hooks/useIsSafeOwner'
+import type { Message } from '@/hooks/useMessages'
+
+const useIsMessageSignableBy = (message: Message, walletAddress: string): boolean => {
```
Comment (schmanu) [inline src/hooks/useIsMsgSignableBy.ts:4]: Lets add a test for this hook

File: src/components/messages/MsgSummary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      {!!confirmationsRequired && (
+        <Box gridArea="confirmations" display="flex" alignItems="center" gap={1}>
+          <TxConfirmations
+            submittedConfirmations={confirmationsSubmitted}
+            requiredConfirmations={confirmationsRequired}
+          />
+        </Box>
+      )}
+
+      {wallet && !isWrongChain && !isConfirmed && (
```
Comment (schmanu) [inline src/components/messages/MsgSummary/index.tsx:61]: Some descisions like 
- is the wallet an owner
- Is the message already signed by the wallet 

are done inside the `SignMsgButton` and render it disabled while some conditions like
- `isWrongChain`
-  no wallet connected 
hide it entirely. 
How exactly do we decide on this? Shouldn't we just move the `isWrongChain` into `useIsMsgSignable`?

File: src/components/messages/PaginatedMsgs/index.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import MsgList from '@/components/messages/MsgList'
+import SkeletonTxList from '@/components/common/PaginatedTxns/SkeletonTxList'
+
+const NoMessages = (): ReactElement => {
+  return (
+    <PagePlaceholder
+      img={<NoMessagesIcon />}
+      text={
+        <Typography variant="body1" color="primary.light" m={2} maxWidth="600px">
+          Some applications allow you to interact with then via off-chain contract signatures (&ldquo;messages&ldquo;)
```
Comment (schmanu) [inline src/components/messages/PaginatedMsgs/index.tsx:21]:
```suggestion
          Some applications allow you to interact with them via off-chain contract signatures (&ldquo;messages&ldquo;)
```

File: src/components/messages/PaginatedMsgs/index.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+          that you can generate with your Safe.
+        </Typography>
+      }
+    >
+      {/* TODO: Add link to help article */}
```
Comment (schmanu) [inline src/components/messages/PaginatedMsgs/index.tsx:26]: I assume the help article is still WIP?

Comment (iamacook) [reply src/components/messages/PaginatedMsgs/index.tsx:26]: Yes.

File: src/components/messages/MsgSummary/index.tsx (hunk 4 above)
Comment (iamacook) [reply src/components/messages/MsgSummary/index.tsx:61]: As discussed, this logic mirrors [that currently implemented in `TxSummary`](https://github.com/safe-global/web-core/blob/dev/src/components/transactions/TxSummary/index.tsx#L96).

File: src/hooks/__tests__/useIsMessageSignableBy.test.ts [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+      ],
+    } as Message
+
+    const address = '0x456'
+
+    const { result } = renderHook(() => useIsMessageSignableBy(message, address))
+    expect(result.current).toBe(false)
+  })
+
+  it('returns false if the message if not previously signed', () => {
```
Comment (schmanu) [inline src/hooks/__tests__/useIsMessageSignableBy.test.ts:69]:
```suggestion
  it('returns false if the message is not previously signed', () => {
```

**Review by schmanu — APPROVED** [review]
There is just one small typo. I'll already approve. Good work!

**Pull request merged**